layouts/partials/head.html

```diff
@@ -15,33 +15,40 @@
             var startYearMonth = startStr.slice(0, 7);
             var endYearMonth = endStr.slice(0, 7);
 
-            // Prepare promises to fetch event data for all months in the range
-            var promises = [];
-            for (
-              var date = new Date(startYearMonth);
-              date <= new Date(endYearMonth);
-              date.setMonth(date.getMonth() + 1)
-            ) {
-              var yearMonth = date.toISOString().slice(0, 7);
-              promises.push(
-                fetch("/events/" + yearMonth + ".json")
-                  .then((res) => res.json())
-                  .catch(() => [])
-              );
-            }
+            // Fetch the months with events.
+            fetch("/events/events.json")
```

Comment:
Although this eliminates the 404 errors, I'm not a fan of this solution as it requires an additional network request to fetch the months, before fetching the actual data.

I do like where you were heading with creating a list for all the months and only making a request if we know that there are events that month.

We know that information at build time, so perhaps we could use that to our advantage and have that list injected in to here:

https://github.com/zidhuss/muscat-tech.org/blob/8820ed6ac893180957b2692124f74c867c59b782/layouts/partials/head.html#L25-L31

For me personally, like I mentioned in the issue, the 404 errors don't bother me that much as they're expected. However I'm all for eliminating them with the right approach.

Reply:
The inject approach I suggested above probably adds more complexity than is necessary. My main point is we shouldn’t have to make 2 requests when changing months.

Reply:
Yes, I do agree with you. I've changed the approach a little bit.